docs/docs/sourcing-from-agilitycms.md

```diff
@@ -46,52 +50,274 @@ Install the dependencies:
+          <AgilityPageTemplate {...viewModel} />
+        </main>
+        <SiteFooter />
+      </div>
+    </>
+  )
+}
+
+export default AgilityPage
+```
+
+[Learn more about How Agility Pages Work in Gatsby](https://help.agilitycms.com/hc/en-us/articles/360043035211)
+
+### How Page Modules Work
+
+Page Modules in Agility CMS are the functional components that make up a page. Editors use these to compose what type of content is on each page and in what order they appear. Developers define what page modules are available in the CMS and what fields they have. Each module defined in Agility CMS should correspond to a React Component in your Gatsby site.
+
+In the [agilitycms-gatsby-starter](https://github.com/agility/agilitycms-gatsby-starter) site, the name of the page module is used to find a corresponding react component that matches the same name. If a match is found, that component is dynamically imported and rendered.
+
+For example, if a module has a reference name of RichTextArea in the CMS, then while the page is being rendered by the gatsby-source-agilitycms plugin, it will look for `./src/components/agility-pageModules/RichTextArea.jsx` in the `./src/components/agility-pageModules` directory.
+
+**RichTextArea.jsx:**
+
+```shell
+import React from "react"
+import { renderHTML } from "../../agility/utils"
+
+const RichTextArea = ({ module }) => {
+  // get module fields
+  const { customFields } = module
+
+  return (
+    <div className="relative px-8">
+      <div className="max-w-2xl mx-auto my-12 md:mt-18 lg:mt-20">
+        <div
+          className="prose max-w-full mx-auto"
+          dangerouslySetInnerHTML={renderHTML(customFields.textblob)}
+        />
+      </div>
+    </div>
+  )
+}
+
+export default RichTextArea
+```
+
+[Learn more about How Agility Page Modules Work in Gatsby](https://help.agilitycms.com/hc/en-us/articles/360043037491)
+
+### How Page Templates Work
+
+Page Templates in Agility CMS are how developers can differentiate the styles of certain types of pages, and define where on the page that editors can add Modules (functional components of the page that editors control). Some sites may only have a single page template and this is re-used across the site, others may have other templates to allow for more flexible layouts.
+
+In the [agilitycms-gatsby-starter](https://github.com/agility/agilitycms-gatsby-starter) site, the Name of the Page Template is used to find a corresponding react component that matches the same name. If a match is found, that component is dynamically imported and rendered.
 
-All of those pages and content are then made available in GraphQL to the React Components you will write to render those pages.
+For example, if a Page Template has a reference name of MainTemplate in the CMS, then while the page is being rendered by the gatsby-source-agilitycms plugin, it will look for ./src/components/agility-pageTemplates/MainTemplate.jsx in the ./src/components/agility-pageTemplates directory.
```

Comment:
```suggestion
For example, if a Page Template has a reference name of MainTemplate in the CMS, then while the page is being rendered by the `gatsby-source-agilitycms` plugin, it will look for `src/components/agility-pageTemplates/MainTemplate.jsx` in the `src/components/agility-pageTemplates` directory.
```